rafthttp/stream.go

```diff
@@ -380,6 +400,18 @@ func (cr *streamReader) dial() (io.ReadCloser, error) {
 		return nil, err
 	case http.StatusOK:
 		return resp.Body, nil
+	case http.StatusNotFound:
+		b, err := ioutil.ReadAll(resp.Body)
```

Comment:
etcd 2.0 mixes two cases together, and etcd 2.1 needs to distinguish them through response body.